src/lib/y2storage/partitionable.rb

```diff
@@ -262,11 +262,20 @@ def preferred_ptable_type
     # Returns the partition table, creating an empty one if needed.
     # @see #preferred_ptable_type
     #
-    # @return [PartitionTable]
+    # @return [PartitionTables::Base]
     def ensure_partition_table
       partition_table || create_partition_table(preferred_ptable_type)
     end
 
+    # Deletes the partition table
+    #
+    # @note All dependent devices will be deleted (MD RAIDs, LVM, etc).
+    def delete_partition_table
```

Comment:
It would be nice to be more consistent with these names.

On one hand, we have `Device#remove_descendants`, `Mountable#remove_mount_point`, `Md#remove_device`, `LvmVg#remove_lvm_pv` or `BlkDevice#remove_encryption`.

On the other hand, we have `PartitionTables#Base.delete_partition`, `PartitionTables#Base.delete_all_partitions` and `LvmVg#delete_lvm_lv`, `BlkDevice#delete_filesystem`.

Not something for this PR, I know... but cases like `BlkDevice#remove_encryption` vs `BlkDevice#delete_filesystem` makes me wanna cry. Looks like the PHP API.

Reply:
Yes, I also hate it. In this case I have to use on of both options, and the closer was `#delete_partition`, so ...